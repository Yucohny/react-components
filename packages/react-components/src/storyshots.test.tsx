import path from 'path'
import initStoryshots from '@storybook/addon-storyshots'

jest.mock('react-dom', () => ({
  createPortal: (node: any) => <div data-create-portal={true}>{node}</div>,
}))

initStoryshots({
  configPath: path.resolve(__dirname, '../configs/storybook'),
})
