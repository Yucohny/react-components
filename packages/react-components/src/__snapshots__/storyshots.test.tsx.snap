// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Html base 1`] = `
<div
  style={
    Object {
      "color": "var(--font-color)",
    }
  }
>
  Some text
</div>
`;

exports[`Storyshots Popover State Up 1`] = `
<div>
  <div
    className="Popover__trigger "
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <p
      style={
        Object {
          "background": "#ddd",
        }
      }
    >
      Trigger
    </p>
  </div>
  <div
    data-create-portal={true}
  >
    <div
      className="Popover__content"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </div>
</div>
`;

exports[`Storyshots Popover base 1`] = `
<div>
  <div
    className="Popover__trigger "
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <p
      style={
        Object {
          "background": "#ddd",
        }
      }
    >
      Trigger
    </p>
  </div>
  <div
    data-create-portal={true}
  >
    <div
      className="Popover__content"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    />
  </div>
</div>
`;

exports[`Storyshots Portal base 1`] = `
<div>
  <p>
    <label>
      closeOnClick
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </label>
  </p>
  <p>
    <label>
      closeOnOutsideClick
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
    </label>
  </p>
  <p>
    <button
      onClick={[Function]}
      type="button"
    >
      Show Portal
    </button>
  </p>
  <div
    data-create-portal={true}
  >
    <div
      className="PortalContent__overlay"
    >
      <div
        className="PortalContent__content"
      >
        <p>
          Portal content
        </p>
        <p>
          <button
            onClick={[Function]}
            type="button"
          >
            Close Portal
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;
